### ISSUE TITLE
aria-label is seen in select and button tag

### ISSUE THREAD

**Issue by SairamAlagapan** [catamphetamine/react-responsive-ui#8]
Hello, I have an issue with aria-label. When I do the screenreader testing with VoiceOver I am able to hear the aria-label being read twice since it's available in both select and button tag. Is there anyway to remove aria-label from select tag or please provide other suggestion if there's any. Find the screenshot for your reference.
![Screen Shot 2022-03-15 at 2 49 25 PM](https://user-images.githubusercontent.com/59094456/158450483-accd7fba-177d-4326-a3c6-c5491c8034c8.png)

https://catamphetamine.github.io/react-responsive-ui/#select

**Comment by catamphetamine** [comment]
Hi.
I don't have any duplication when testing with NVDA on Windows.
The `<button/>` doesn't get focused via Tab key because it has `tabIndex={-1}`.
Therefore, VoiceOver shouldn't announce its `aria-label`.

If we removed `aria-label` on `<select/>`, then it's not clear what it would have announced when focused.
You could test that by removing the attribute in the Dev Console and then testing with VoiceOver.

**Comment by SairamAlagapan** [comment]
The SR user uses Control+Option+Arrow Keys to read the contents. When you do that using VoiceOver, it reads out twice. I tried removing `aria-label` on `<select/>` in the dev console and it read out only once but, when I try removing it from my code, the `aria-label` gets removed from both `<select/>` and `<button/>` tags. Any suggestions?

**Comment by catamphetamine** [comment]
> The SR user uses Control+Option+Arrow Keys to read the contents. 

So it focuses the `<button/>`? Should it? Having `tabIndex={-1}` there. Perhaps it shouldn't.

> I try removing it from my code

Not from your code. From the Dev Tools panel's Elements tab.

**Comment by SairamAlagapan** [comment]
So it focuses the `<button/>`? Should it? Having `tabIndex={-1}` there. Perhaps it shouldn't. - Yes it focuses the button.

I removed aria-label from the `<select/>` tag from dev tools panel's elements tab and it's reading out only once. Is there anyway you can do the changes in your code so that I can use it.

**Comment by SairamAlagapan** [comment]
Hello,

Any updates on this?

**Comment by catamphetamine** [comment]
@SairamAlagapan Oh, didn't notice your comment.

> Yes it focuses the button.

That's strange. Its `tabIndex` is `-1` — it's strange if it focuses the button.

> I removed aria-label from the `<select/>` tag from dev tools panel's elements tab and it's reading out only once. 

Specify what exactly does it read out in both cases.

> Is there anyway you can do the changes in your code so that I can use it.

I think the browser might be somehow "misbehaving" in this case

**Comment by SairamAlagapan** [comment]
Please find the video for reference. It reads out as menu pop up, collapsed, button when there is no aria-label in the `<select/>` tag.

https://user-images.githubusercontent.com/59094456/160180361-5c79641c-1263-4f4d-a5e9-a3467a0e91b8.mov

I want like this for my problem to be resolved. Only one aria-label need to be announced.

**Comment by catamphetamine** [comment]
Hmm, I see.

In the latest version, it supports an `id` property on the `<Select/>` component.
One could use that `id` property to link a `<label/>` to the `<select/>` element:

```html
<label for="selectId">Fruit</label>
<Select id="selectId" options=.../>
```

Another option would be supplying an `aria-label` property:

```html
<Select id="selectId" aria-label="Fruit" options=.../>
```

As for why does it focus the `<button/>` when tabbing through — I dunno.
The `<button/>` is marked as `tabIndex={-1}` so it shouldn't get focused on Tab.

I also don't have a Mac to test on.
You could attempt to find a way to disable VoiceOver reading out the `<button/>` yourself.

**Comment by SairamAlagapan** [comment]
It doesn't focus the `<button/>` when tabbing. It focuses the button only when I turn on the voiceover and use Control + Option + Arrow Keys.

Thank you for the suggestion. I will try it from my side and will ask you if I face any challenges.

**Comment by catamphetamine** [comment]
> It focuses the button only when I turn on the voiceover and use Control + Option + Arrow Keys.

I see. Well, perhaps there could be a way to hide this `<button/>` from VoiceOver?
Maybe adding `aria-hidden` HTML attribute would work?
You could test it in the Dev Tools.

**Comment by catamphetamine** [comment]
Published `react-responsive-ui@0.15.26` with the `aria-hidden` workaround.

**Comment by SairamAlagapan** [comment]
Hello @catamphetamine,

I pulled the latest changes and it's working fine now. Thank you for your response and action.
